source/guide_up1.rst

```diff
@@ -0,0 +1,217 @@
+ [isabell@stardust ~]$
+ 
+We will need to update the Node Packet Manager npm:
+
+::
+
+  [isabell@stardust ~]$ npm install npm@latest -g
+  + npm@6.4.1
+  updated 1 package in 14.629s
+  [isabell@stardust ~]$ 
+
+
+.. include:: includes/my-print-defaults.rst
+
+Your URL needs to be setup:
+
+.. include:: includes/web-domain-list.rst
+
+Installation
+============
+
+First get the Up1 source code from Github_:
+
+::
+
+  [isabell@stardust ~]$ git clone https://github.com/Upload/Up1 up1 --depth 1
+  Cloning into 'up1'...
+  remote: Enumerating objects: 37, done.
+  remote: Counting objects: 100% (37/37), done.
+  remote: Compressing objects: 100% (33/33), done.
+  remote: Total 37 (delta 0), reused 16 (delta 0), pack-reused 0
+  Unpacking objects: 100% (37/37), done.
+  Note: checking out '059f199a6f94fc6c90cde1e4dab0b5b85b49533c'.
+  [isabell@stardust ~]$
+
+You have now downloaded the master development branch, because this might behave like a beta version of the software, you should check out the latest release. Please be sure to replace the pseudo version number ``66.6`` here with the latest version number from the release feed_:
+
+::
+
+  [isabell@stardust ~]$ cd ~/up1
+  [isabell@stardust up1]$ git checkout v66.6 --depth1
+  You are in 'detached HEAD' state. You can look around, [...]
+  [isabell@stardust up1]$ git checkout -b latest
+  Switched to a new branch 'latest'
+  [isabell@stardust up1]$
+
+Then run npm_ to install all the dependencies:
+
+::
+
+  [isabell@stardust ~]$ cd ~/up1/server
+  [isabell@stardust server]$ npm install
+  [...]
+  added 131 packages from 90 contributors and audited 219 packages in 6.04s
+  found 17 vulnerabilities (4 low, 8 moderate, 5 high)
+  run `npm audit fix` to fix them, or `npm audit` for details
+  [isabell@stardust server]$ 
+
+You probably got the error that there are vulnerabilities to fix, so follow the instruction and run ``npm audit fix`` to install the latest package versions:
+
+::
+
+  [isabell@stardust ~]$ cd ~/up1/server
+  [isabell@stardust server]$ npm audit fix
+  [...]
+  fixed 17 of 17 vulnerabilities in 219 scanned packages
+  [isabell@stardust server]$
+
+Configuration
+=============
+
+Configure port
+--------------
+
+Since Node.js applications use their own webserver, you need to find a free port and bind your application to it.
+
+.. include:: includes/generate-port.rst
+
+Change the configuration
+------------------------
+
+Copy the example settings files to create your own configuration:
+
+::
+
+  [isabell@stardust ~]$ cd ~/up1
+  [isabell@stardust up1]$ cp server/server.conf.example server/server.conf
+  [isabell@stardust up1]$ cp client/config.js.example server/config.js
+  [isabell@stardust up1]$
+
+Now edit ``~/up1/server/server.conf`` and replace ``<yourport>`` with the fresh portnumber you got earlier. Also set a random string for ``api_key`` and ``delete_key``:
+
+.. code-block:: none
+ :emphasize-lines: 2,3,13,18
+
+  {
+  "api_key": "<any_random_string>",
```

Comment:
Maybe use something like `< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c${1:-32};echo;` to generate the random strings?

Reply:
I was not sure if the guide should tell the people where they get there random strings from? If you think its ok, I can add it this way.

Reply:
Hm. I think it's better to give them a hint, otherwise many people probably leave this 'as is'.